lib/worker/exec.go

```diff
@@ -0,0 +1,129 @@
+package worker
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// Exec represents something to be executed in the Worker
+// TODO: I might just remove this interface in the next
+// PR. I introduced it thinking it could be useful for
+// the future cgroup implementation but now am not so
+// sure.
+type Exec interface {
+	// Command is the command for the exec.
+	Command() string
+	// Start starts the exec.
+	Start() error
+	// Stop stops the exec.
+	Stop() error
+	// PID returns the pid of the process once it has been started.
+	// Otherwise, and invalid pid 0 is returned.
+	PID() int
+	// Status returns the ProcessStatus struct corresponding
+	// to the started exec.
+	Status() (*ProcessStatus, error)
+}
+
+// process is a process executed or to be executed by the worker
+// TODO: Find a better name for this as the word "process" implies
+// that this it is already running.
+type process struct {
+	command    string
+	cmd        *exec.Cmd
+	startedBy  string
+	startedAt  time.Time
+	finishedAt time.Time // TODO: implement finished at
```

Comment:
What will this be for an in-flight process? Zero value? And how do you know when it's valid?